### PR TITLE
Add replacement Strategy to schema

### PR DESCRIPTION
**Pull request by wenyhu**
*Issue #, if available:*
https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/pull/86

*Description of changes:*
Description of changes: Allow replacement Strategy in the schema. This allows immutable singleton resources to update in the order of delete then create.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ammokhov — APPROVED** [review]

File: src/rpdk/core/data/schema/provider.definition.schema.v1.json [hunk 1]
```diff
@@ -71,6 +71,17 @@
                 }
             }
         },
+        "replacementStrategy": {
+            "type": "string",
+            "description": "The valid replacement strategies are [create_then_delete] and [delete_then_create]. All other inputs are invalid.",
+            "default": [
```
Comment (jsennett) [inline src/rpdk/core/data/schema/provider.definition.schema.v1.json:77]: Should this be "default": "create_then_delete" (not an array)? 

I am developing a new resource type that could use this property and am looking forward to this new feature!

Comment (wenyhu) [reply src/rpdk/core/data/schema/provider.definition.schema.v1.json:77]: Thanks a lot for the good catch! Fixing it.

Comment (jsennett) [reply src/rpdk/core/data/schema/provider.definition.schema.v1.json:77]: No problem. I wasn't referring to the description though, the default value is still an array:
"default": ["create_then_delete"]     ----> "default": "create_then_delete"

Comment (wenyhu) [reply src/rpdk/core/data/schema/provider.definition.schema.v1.json]: Fixed that. Thanks!

**Review by wbingli — APPROVED** [review]

**Pull request merged**